pyHalo/realization_extensions.py

```diff
@@ -377,26 +378,36 @@ def add_primordial_black_holes(self, pbh_mass_fraction, kwargs_pbh_mass_function
             delta_z.append(plane_redshifts[i + 1] - plane_redshifts[i])
         delta_z.append(self._realization.lens_cosmo.z_source - plane_redshifts[-1])
 
-        geometry = Geometry(self._realization.lens_cosmo.cosmo,
+        geometry1 = Geometry(self._realization.lens_cosmo.cosmo,
```

Comment:
I think you should just create a new instance of Geometry inside the for loop

Reply:
That will just make it slower. If you want it to be more elegant, I could make it a flag or something, but I don't want to just do unnecessary operations in the loop

Reply:
it takes on average 280 micro sec to create an instance of Geometry so it shouldn't be an issue. flags and additional if statements make the code less readable and I want to avoid that

Reply:
okay, I had a slowdown problem with creating multiple instances before but it was probably much more than four, so I'll change it.